src/examples/server/simpleFastifyServer.ts

```diff
@@ -0,0 +1,211 @@
+import Fastify from "fastify";
+import { McpServer } from "../../server/mcp.js";
+import { StreamableHTTPServerTransportOptions } from "../../server/streamableHttp.js";
+import { RawHttpServerAdapter } from "../../server/raw-http-adapter.js";
+import { randomUUID } from "node:crypto";
+import { z } from "zod";
+import { CallToolResult, GetPromptResult, ReadResourceResult } from '../../types.js';
+
+async function runFastifyMCPServer() {
+  const fastify = Fastify({ logger: true });
+
+  const mcpAdapterOptions: StreamableHTTPServerTransportOptions = {
+    sessionIdGenerator: () => randomUUID(),
+    // For true JSON request/response (non-SSE) for POSTs, uncomment the next line.
+    // Otherwise, POST requests that expect a response will use an SSE stream.
+    // enableJsonResponse: true,
+  };
+  const mcpAdapter = new RawHttpServerAdapter(mcpAdapterOptions);
+
+  const mcpServer = new McpServer({
+    name: "simple-streamable-http-server",
+    version: "1.0.0",
+  }, { capabilities: { logging: {} } });
+
+  // === Tools, Resource, and Prompt from simpleStreamableHttp.ts ===
+
+  // Tool: greet
+  mcpServer.tool(
+    'greet',
+    'A simple greeting tool',
+    {
+      name: z.string().describe('Name to greet'),
+    },
+    async ({ name }): Promise<CallToolResult> => {
+      return {
+        content: [
+          {
+            type: 'text',
+            text: `Hello, ${name}!`,
+          },
+        ],
+      };
+    }
+  );
+
+  // Tool: multi-greet
+  mcpServer.tool(
+    'multi-greet',
+    'A tool that sends different greetings with delays between them',
+    {
+      name: z.string().describe('Name to greet'),
+    },
+    {
+      title: 'Multiple Greeting Tool', 
+      readOnlyHint: true,
+      openWorldHint: false
+    },
+    async ({ name }, { sendNotification }): Promise<CallToolResult> => {
+      const sleep = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+      await sendNotification({
+        method: "notifications/message",
+        params: { level: "debug", data: `Starting multi-greet for ${name}` }
+      });
+
+      await sleep(1000); // Wait 1 second before first greeting
```

Comment:
Minor nitpick, but this library targets NodeJS 18+, so you should be able to safely use `await setTimeout(1000);` here instead of implementing a sleep function.

See [documentation](https://nodejs.org/docs/latest-v18.x/api/timers.html#timers-promises-api) for more info

Reply:
you're right !
done :)